### ISSUE TITLE
issue with lazy.restart() and lazy.shutdown()

### ISSUE THREAD

**Issue by srt0** [qtile/qtile#1609]
# Issue description

1. When I use lazy.shutdown(), sometimes it got stuck. Nothing on screen except the mouse pointer. No key binding works.

2. When I use lazy.restart(), BAR doesn't come back and no key binding works or acting as lazy.shutdown().

# Qtile version

Qtile v 0.15.0

# Stack traces

Please attach any stack traces found in:

* `~/.local/share/qtile/qtile.log`

2020-04-13 17:36:41,594 ERROR libqtile xcore.py:_xpoll():L277 Got an exception in poll loop
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 246, in _xpoll
    ret = target(event)
  File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 494, in handle_KeyPress
    self.qtile.process_key_event(keysym, event.state & self._valid_mask)
  File "/usr/lib/python3.8/site-packages/libqtile/core/manager.py", line 336, in process_key_event
    key = self.keys_map[(keysym, mask)]
KeyError: (65515, 0)
2020-04-13 17:41:19,226 WARNING libqtile xcbq.py:set_property():L687 X error in SetProperty (wid=14680070, prop='WM_STATE'), ignoring
2020-04-13 17:43:43,697 WARNING libqtile xcursors.py:_setup_xcursor_binding():L112 xcb-cursor not found, fallback to font pointer
2020-04-13 17:43:44,476 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.CPUGraph', No module named 'psutil'
2020-04-13 17:43:44,478 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.MemoryGraph', No module named 'psutil'
2020-04-13 17:43:44,479 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.SwapGraph', No module named 'psutil'
2020-04-13 17:43:44,480 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.NetGraph', No module named 'psutil'
2020-04-13 17:43:44,482 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.HDDGraph', No module named 'psutil'
2020-04-13 17:43:44,483 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.HDDBusyGraph', No module named 'psutil'
2020-04-13 17:43:45,315 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.mpd2widget.Mpd2', No module named 'mpd'
2020-04-13 17:43:45,446 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.wlan.Wlan', No module named 'iwlib'
2020-04-13 17:43:45,454 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.khal_calendar.KhalCalendar', No module named 'dateutil'
2020-04-13 17:43:45,458 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.imapwidget.ImapWidget', No module named 'keyring'
2020-04-13 17:43:45,462 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.net.Net', No module named 'psutil'
2020-04-13 17:43:45,495 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.memory.Memory', No module named 'psutil'
2020-04-13 17:43:45,499 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.cpu.CPU', No module named 'psutil'
2020-04-13 17:48:34,132 WARNING libqtile xcbq.py:set_property():L687 X error in SetProperty (wid=14680070, prop='WM_STATE'), ignoring
2020-04-13 17:48:38,834 WARNING libqtile xcbq.py:set_property():L687 X error in SetProperty (wid=14680070, prop='WM_STATE'), ignoring
2020-04-13 17:48:49,069 WARNING libqtile xcursors.py:_setup_xcursor_binding():L112 xcb-cursor not found, fallback to font pointer
2020-04-13 17:48:49,285 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.CPUGraph', No module named 'psutil'
2020-04-13 17:48:49,287 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.MemoryGraph', No module named 'psutil'
2020-04-13 17:48:49,288 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.SwapGraph', No module named 'psutil'
2020-04-13 17:48:49,289 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.NetGraph', No module named 'psutil'
2020-04-13 17:48:49,290 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.HDDGraph', No module named 'psutil'
2020-04-13 17:48:49,291 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.graph.HDDBusyGraph', No module named 'psutil'
2020-04-13 17:48:49,340 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.mpd2widget.Mpd2', No module named 'mpd'
2020-04-13 17:48:49,347 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.wlan.Wlan', No module named 'iwlib'
2020-04-13 17:48:49,348 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.khal_calendar.KhalCalendar', No module named 'dateutil'
2020-04-13 17:48:49,349 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.imapwidget.ImapWidget', No module named 'keyring'
2020-04-13 17:48:49,351 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.net.Net', No module named 'psutil'
2020-04-13 17:48:49,354 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.memory.Memory', No module named 'psutil'
2020-04-13 17:48:49,355 WARNING libqtile utils.py:safe_import():L191 Unmet dependencies for optional Widget: '.widget.cpu.CPU', No module named 'psutil'

# Configuration

[Qtile Config](https://gist.github.com/srt0/bac949befd6fe02c11cfd6cb27b6f9e6)

**Comment by pawamoy** [comment]
Interesting part seems to be

```
2020-04-13 17:36:41,594 ERROR libqtile xcore.py:_xpoll():L277 Got an exception in poll loop
Traceback (most recent call last):
File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 246, in _xpoll
ret = target(event)
File "/usr/lib/python3.8/site-packages/libqtile/backend/x11/xcore.py", line 494, in handle_KeyPress
self.qtile.process_key_event(keysym, event.state & self._valid_mask)
File "/usr/lib/python3.8/site-packages/libqtile/core/manager.py", line 336, in process_key_event
key = self.keys_map[(keysym, mask)]
KeyError: (65515, 0)
```

The rest is just missing dependencies:

```
pip install psutil iwlib keyring python-dateutil python-mpd
```

and

```
sudo pacman -S xcb-util-cursor
```

**Comment by srt0** [comment]
@pawamoy , thanks for your reply. these resolves the import related issues. but issue with lazy.restart() adn lazy.shutdown() still persists.

**Comment by pawamoy** [comment]
You're welcome :slightly_smiling_face:  Yes, unfortunately I'm not very familiar with the code. It seems the `Super_L` key was not processed correctly? `self.keys_map` contained no `(65515, 0)` key.

**Comment by ramnes** [comment]
Related to #1615 most likely?

**Comment by srt0** [comment]
Issue solved after Qtile updated to v 0.15.1
